### PR TITLE
268 ability to unlock a respondents account

### PR DESCRIPTION
**Pull request by shahi645**
# Motivation and Context
Adding the ability for an internal user to unlock a respondent from RU page on ROps

# What has changed
When respondent is locked out, and RU page is displayed internally. Lock status show with an unlock link below. When clicked a confirm page is displayed. 

# How to test?
Run in conjunction with other branches:
Frontstage: 268-update-party-service-with-acount-lock
ROps: 268-ability-to-unlock-a-respondents-account
Party: 268-party-lock-account-in-djano-oauth2

Try unlocking from ROps, should work and be able to log in.

### CODE REVIEW

File: response_operations_ui/controllers/reporting_units_controllers.py [hunk 1]
```diff
@@ -49,6 +49,26 @@ def change_enrolment_status(business_id, respondent_id, survey_id, change_flag):
                  business_id=business_id, respondent_id=respondent_id, survey_id=survey_id, change_flag=change_flag)
 
 
+def change_respondent_status(respondent_id, change_flag):
+    if change_flag == 'UNLOCKED':
+        change_flag = 'ACTIVE'
```
Comment (tej99) [inline response_operations_ui/controllers/reporting_units_controllers.py]: Why not just give 'ACTIVE' as the value for change_flag param?

File: response_operations_ui/controllers/reporting_units_controllers.py [hunk 2]
```diff
@@ -49,6 +49,26 @@ def change_enrolment_status(business_id, respondent_id, survey_id, change_flag):
...
+        url = f'{app.config["PARTY_URL"]}/party-api/v1/respondents/edit-account-status/{respondent_id}'
+        enrolment_json = {
+            'respondent_id': respondent_id,
+            'status_change': change_flag
+        }
+        response = requests.put(url, json=enrolment_json, auth=app.config['PARTY_AUTH'])
+
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError:
```
Comment (tej99) [inline response_operations_ui/controllers/reporting_units_controllers.py]: `from requests.exceptions import HTTPError` instead?

File: response_operations_ui/controllers/reporting_units_controllers.py [hunk 3]
```diff
@@ -49,6 +49,26 @@ def change_enrolment_status(business_id, respondent_id, survey_id, change_flag):
...
+            logger.error('Failed to change respondent status', respondent_id=respondent_id, change_flag=change_flag)
+            raise ApiError(response)
+
+        logger.info('Successfully changed respondent status', respondent_id=respondent_id, change_flag=change_flag)
```
Comment (tej99) [inline response_operations_ui/controllers/reporting_units_controllers.py]: debug log level?

File: response_operations_ui/templates/reporting-unit.html [hunk 4]
```diff
@@ -170,14 +170,23 @@ <h4 class="neptune">Respondents</h4>
               </dl>
             </td>
             <td name="tbl-respondent-status" class="table--cell">
-              {{ respondent.status.title() }}
+              {% if respondent.status  == 'SUSPENDED' %} Locked {% else %} {{ respondent.status.title() }} {% endif %}
+              <div>
+{#                {% if respondent.status == 'ACTIVE' %}#}
+{#                <a href="{{ url_for('reporting_unit_bp.confirm_change_respondent_status', ru_ref=ru.sampleUnitRef, respondent_first_name=respondent.firstName,#}
+{#                                    respondent_last_name=respondent.lastName, email_address=respondent.emailAddress, change_flag='LOCKED')}}"#}
+{#                   id="change-respondent-status">Lock</a>#}
```
Comment (tej99) [inline response_operations_ui/templates/reporting-unit.html]: What are the `{#` and `#}` for?

File: response_operations_ui/templates/reporting-unit.html [hunk 5]
```diff
@@ -170,14 +170,23 @@ <h4 class="neptune">Respondents</h4>
...
+                {% if respondent.status == 'SUSPENDED' %}
+                <a href="{{ url_for('reporting_unit_bp.confirm_change_respondent_status', ru_ref=ru_ref, party_id=respondent.id, respondent_first_name=respondent.firstName,
+                                    respondent_last_name=respondent.lastName, email_address=respondent.emailAddress, change_flag='UNLOCKED') }}">Unlock</a>
```
Comment (tej99) [inline response_operations_ui/templates/reporting-unit.html]: We shouldn't be passing email as a query param

File: response_operations_ui/views/reporting_units.py [hunk 6]
```diff
@@ -225,6 +227,18 @@ def confirm_change_enrolment_status(ru_ref):
...
 
+@reporting_unit_bp.route('/<ru_ref>/change-respondent-status', methods=['GET'])
+@login_required
+def confirm_change_respondent_status(ru_ref):
+    return render_template('confirm-respondent-status-change.html',
+                           ru_ref=ru_ref,
+                           respondent_id=request.args['party_id'],
+                           first_name=request.args['respondent_first_name'],
+                           last_name=request.args['respondent_last_name'],
+                           email_address=request.args['email_address'],
```
Comment (tej99) [inline response_operations_ui/views/reporting_units.py]: We shouldn't be passing email as a query param

File: response_operations_ui/templates/reporting-unit.html (hunk 4 above)
Comment (shahi645) [reply response_operations_ui/templates/reporting-unit.html]: So this is the ability to `lock the account` with a link but we dont want to implement yet, so I have commented out the html for now

File: response_operations_ui/views/reporting_units.py (hunk 6 above)
Comment (shahi645) [reply response_operations_ui/views/reporting_units.py]: we shouldn't be...

File: response_operations_ui/controllers/reporting_units_controllers.py (hunk 1 above)
Comment (shahi645) [reply response_operations_ui/controllers/reporting_units_controllers.py]: fair play

File: response_operations_ui/controllers/reporting_units_controllers.py [hunk 7]
```diff
@@ -49,6 +50,25 @@ def change_enrolment_status(business_id, respondent_id, survey_id, change_flag):
...
+        url = f'{app.config["PARTY_URL"]}/party-api/v1/respondents/edit-account-status/{respondent_id}'
+        enrolment_json = {
+            'respondent_id': respondent_id,
+            'status_change': change_flag
+        }
+        response = requests.put(url, json=enrolment_json, auth=app.config['PARTY_AUTH'])
+
+        try:
+            response.raise_for_status()
+        except HTTPError:
```
Comment (MIslam92) [inline response_operations_ui/controllers/reporting_units_controllers.py]: Except part not covered in tests?

File: response_operations_ui/views/reporting_units.py [hunk 8]
```diff
@@ -225,6 +228,19 @@ def confirm_change_enrolment_status(ru_ref):
                            change_flag=request.args['change_flag'])
 
 
+@reporting_unit_bp.route('/<ru_ref>/change-respondent-status', methods=['GET'])
+@login_required
+def confirm_change_respondent_status(ru_ref):
+    respondent = get_respondent_by_party_id(request.args['party_id'])
```
Comment (MIslam92) [inline response_operations_ui/views/reporting_units.py]: Not covered up in test

**Pull request merged**